clippy_lints/src/strings.rs

```diff
@@ -191,7 +191,7 @@ impl<'tcx> LateLintPass<'tcx> for StringAdd {
             },
             ExprKind::Index(target, _idx, _) => {
                 let e_ty = cx.typeck_results().expr_ty(target).peel_refs();
```

Comment:
Can we just directly call `expr_ty_adjusted` here? That seems like it should cover both `&str` and `Cow<str>` in one and shouldn't need the extra check

Reply:
Thanks for simplification hint!